test/conversions.jl

```diff
@@ -55,6 +55,13 @@ module TestConversions
     @test df[:b] == b
     @test df[:c] == c
 
+    df = invoke(convert, (Type{DataFrame}, Associative), DataFrame, di)
```

Comment:
Why not test the actual use the case this PR fixes, and use an `OrderedDict`? You can add DataStructures.jl to `test/REQUIRES`, it's not a terribly heavy dependency.